basyx.components/basyx.components.docker/basyx.components.AASServer/src/main/java/org/eclipse/basyx/components/aas/mongodb/MongoDBSubmodelAPI.java

```diff
@@ -559,7 +562,7 @@ public java.io.File getSubmodelElementFile(String idShortPath) {
 			File fileSubmodelElement = File.createAsFacade(submodelElement);
 			GridFSBucket bucket = getGridFSBucket();
 			String fileName = constructFileName(fileSubmodelElement, idShortPath);
-			java.io.File file = new java.io.File(fileName);
+			java.io.File file = new java.io.File(tmpDirectory, fileName.replaceAll("[^a-zA-Z0-9-_\\.]", "_"));
```

Comment:
Please move the replaceAll part to the constructFIleName function

Reply:
Done, did it outside before on purpose because I was not 100% sure if placing it inside constructFileName will break something storage related as fileName is also used to identify the file when fetching from MongoDB ...